### PR TITLE
Use Virtio random for entropy in QEMU

### PR DESCRIPTION
**Pull request by arichardson**
When running the minimal disk image I noticed that we were blocking for 20-60 seconds when starting dhclient. It turns out this is due to it calling `arc4random()` when there isn't enough entropy.

I've now added virtio-random as a fast hardware entropy source which makes this problem go away. However, I noticed that the random harvesting thread runs 10 times per second gathering entropy, which is ridiculously often on QEMU. I've updated the thread to suspend when no entropy is requested and resume when `read_rate_increment()` is called.

If we further want to reduce overhead I think we should just use virtio-random directly for generating the random data.

### CODE REVIEW

**Comment by arichardson** [comment]
I know this is quite hacky (replacing the random algorithm pointer) but it does noticeably speed up running the libcxx test suite in QEMU and seems to work just fine.

File: sys/dev/random/random_infra.c [hunk 1]
```diff
@@ -50,9 +50,9 @@ MALLOC_DEFINE(M_ENTROPY, "entropy", "Entropy harvesting buffers and data structu
 struct sources_head source_list = LIST_HEAD_INITIALIZER(source_list);
 
 #if defined(RANDOM_LOADABLE)
-struct random_algorithm *p_random_alg_context = NULL;
+volatile struct random_algorithm *p_random_alg_context = NULL;
 #else /* !defined(RANDOM_LOADABLE) */
-struct random_algorithm *p_random_alg_context = &random_alg_context;
+volatile struct random_algorithm *p_random_alg_context = &random_alg_context;
```
Comment (jrtc27) [inline sys/dev/random/random_infra.c]: Why do these need to be volatile?

**Comment by brooksdavis** [comment]
I feel like this is OBE.

**Comment by arichardson** [comment]
I'm pretty sure the random code is still horrendously slow and does stupid things but this pr can be closed.

**Pull request closed without merging**